upup/pkg/fi/cloudup/bootstrapchannelbuilder/bootstrapchannelbuilder.go

```diff
@@ -870,7 +870,47 @@ func (b *BootstrapChannelBuilder) buildAddons(c *fi.CloudupModelBuilderContext)
 		if b.Cluster.Spec.GetCloudProvider() == kops.CloudProviderGCE {
 			if b.Cluster.Spec.ExternalCloudControllerManager != nil {
 				key := "gcp-cloud-controller.addons.k8s.io"
-				{
+				useBuiltin := !b.hasExternalAddon(key)
+
+				if !useBuiltin {
+					klog.Infof("Found cloud-controller-manager in addons; won't use builtin")
+
+					// Until we make the manifest extensible, we still need to inject our arguments.
+					// TODO(justinsb): we don't really want to do this, it limits the ability for users to override things.
+					// However, this is behind a feature flag at the moment, and this way we can work towards something better.
+					gkDaemonset := schema.GroupKind{Group: "apps", Kind: "DaemonSet"}
+					for _, addon := range b.ClusterAddons {
+						if addon.GroupVersionKind().GroupKind() == gkDaemonset &&
+							addon.GetName() == "cloud-controller-manager" &&
+							addon.GetNamespace() == "kube-system" {
+
+							klog.Infof("replacing arguments in externally provided cloud-controller-manager")
+
+							fnAny, ok := b.templates.TemplateFunctions["CloudControllerConfigArgv"]
+							if !ok {
+								return nil, nil, fmt.Errorf("unable to find TemplateFunction CloudControllerConfigArgv")
+							}
+							fn, ok := fnAny.(func() ([]string, error))
+							if !ok {
+								return nil, nil, fmt.Errorf("unexpected typed for TemplateFunction CloudControllerConfigArgv: %T", fnAny)
```

Comment:
```suggestion
								return nil, nil, fmt.Errorf("unexpected type for TemplateFunction CloudControllerConfigArgv: %T", fnAny)
```

Reply:
Thanks - merged this and the other one, squashed and made you a co-author